src/userdev/java/net/minecraftforge/gradle/userdev/util/DeobfuscatingRepo.java

```diff
@@ -126,12 +126,8 @@ private File findRaw(Artifact artifact, String mapping) throws IOException {
     }
 
     private File findSource(Artifact artifact, String mapping) throws IOException {
-        Optional<File> orig = findArtifactFile(artifact);
-        if (!orig.isPresent()) {
-            return null;
-        }
-
-        File origFile = orig.get();
+        File origFile = MavenArtifactDownloader.manual(project, artifact.getDescriptor(), false);
```

Comment:
So this indeed does solve the problem, but would download the file everytime it is requested. Or at least makes an attempt todo it.

In my eyes it would be better if this is used in combination with the old technique.
In the sense that this new way is used when then optional is empty.

Would that be an option?

Reply:
No, the old mechanic cannot retrieve sources unless a souces classifier artifact is explicitly requested. Same rule apply on pom file.

Reply:
And IDE (idea in my case) would only request sources when it is missing, so it won't make too much network connections. Have no idea how Eclipse attach sources but Lex said Eclipse is working iirc.

Reply:
Okey perfect.
Resolves the issue then.